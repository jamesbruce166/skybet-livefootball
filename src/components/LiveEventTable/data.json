{
	"table-title": "Live Events",
	"empty-table-message": "There is no data to fetch right now, come back later!"
}
