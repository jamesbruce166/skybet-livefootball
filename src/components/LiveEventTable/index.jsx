import React from 'react';

import LiveEventTableCell from '../LiveEventTableCell';
import {
	Section,
	Container,
	EmptyTableIcon,
	EmptyTableCard,
	EmptyTableMessage,
} from './liveEventTable.styles';

import displayData from './data.json';

import { useEvents } from '../../contexts/EventProvider';

const LiveEventsTable = () => {
	const { events } = useEvents();

	const LiveEventsRows = () => {
		const { data } = events;

		return data
			.filter((e) => e.status.displayable)
			.map((event) => {
				const { eventId } = event;
				return <LiveEventTableCell key={eventId} event={event} />;
			});
	};

	const EmptyTable = () => {
		return (
			<EmptyTableCard>
				<EmptyTableIcon />
				<EmptyTableMessage>
					{displayData['empty-table-message']}
				</EmptyTableMessage>
			</EmptyTableCard>
		);
	};

	return (
		<Section data-testid='event-table'>
			<Container>
				{events ? <LiveEventsRows /> : <EmptyTable />}
			</Container>
		</Section>
	);
};

export default LiveEventsTable;
