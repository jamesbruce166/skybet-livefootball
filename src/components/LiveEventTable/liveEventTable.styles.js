import styled from 'styled-components';
import { SECONDARY_DARK } from '../../variables.styles';
import { Football } from '@styled-icons/ionicons-outline';

export const Section = styled.section`
	display: flex;
	min-height: 100vh;
	justify-content: center;
	align-items: center;
	padding: 5%;
`;

export const Container = styled.div`
	width: 80%;
`;

export const EmptyTableCard = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	border-radius: 12px;
	width: 100%;
	min-height: 30vh;
	background: ${SECONDARY_DARK};
`;

export const EmptyTableIcon = styled(Football)`
	color: lightgrey;
	width: 50px;
`;

export const EmptyTableMessage = styled.p`
	color: white;
	font-size: 20px;
	font-weight: 100;
	text-align: center;
`;
