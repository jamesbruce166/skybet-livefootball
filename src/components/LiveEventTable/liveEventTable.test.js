import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';

import displayData from './data.json';
import LiveEventTable from './index';
import mockEvents from './__mocks__/events.json';

afterEach(() => {
	cleanup();
});

test('renders correctly', () => {
	const tree = renderer
		.create(<LiveEventTable events={mockEvents} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

test('should show empty table view if empty', () => {
	const emptyEvents = { data: [] };
	render(<LiveEventTable events={emptyEvents} />);
	const eventTable = screen.queryByTestId('event-table');

	expect(eventTable).toBeInTheDocument();
	expect(eventTable).toHaveTextContent(displayData['empty-table-message']);
});

test('should contain row for each event', () => {
	render(<LiveEventTable events={mockEvents} />);
	const eventTable = screen.queryByTestId('event-table');
	const eventCells = screen.queryAllByTestId('event-cell');
	const cellsRendered = eventCells.length;

	expect(eventTable).toBeInTheDocument();
	expect(cellsRendered).toEqual(mockEvents.data.length);
});
