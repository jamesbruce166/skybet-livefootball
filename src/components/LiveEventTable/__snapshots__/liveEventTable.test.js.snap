// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-ftvSup fDOnoU"
  data-testid="event-table"
>
  <div
    className="sc-papXJ bSmiBp"
  >
    <div
      className="sc-bczRLJ cqpXMw"
      data-testid="event-cell"
    >
      <div
        className="sc-gsnTZi jfffIx"
      >
        <div
          className="sc-dkzDqf caZALy"
        >
          <p
            className="sc-hKMtZM cGURqS"
          >
            Shanghai Shenhua
          </p>
          <p
            className="sc-eCYdqJ bSsXdt"
          >
            0
          </p>
        </div>
        <div
          className="sc-dkzDqf caZALy"
        >
          <p
            className="sc-hKMtZM cGURqS"
          >
            Shandong Luneng Taishan
          </p>
          <p
            className="sc-eCYdqJ bSsXdt"
          >
            0
          </p>
        </div>
      </div>
      <div
        className="sc-jSMfEi fnIhHY"
      >
        <div
          className="sc-iBkjds leGKiQ"
        >
          LIVE
        </div>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 lbJwfL sc-gKXOVf clsDmZ"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="sc-bczRLJ cqpXMw"
      data-testid="event-cell"
    >
      <div
        className="sc-gsnTZi jfffIx"
      >
        <div
          className="sc-dkzDqf caZALy"
        >
          <p
            className="sc-hKMtZM cGURqS"
          >
            Home United
          </p>
          <p
            className="sc-eCYdqJ bSsXdt"
          >
            0
          </p>
        </div>
        <div
          className="sc-dkzDqf caZALy"
        >
          <p
            className="sc-hKMtZM cGURqS"
          >
            Albirex Niigata FC
          </p>
          <p
            className="sc-eCYdqJ bSsXdt"
          >
            1
          </p>
        </div>
      </div>
      <div
        className="sc-jSMfEi fnIhHY"
      >
        <div
          className="sc-iBkjds leGKiQ"
        >
          LIVE
        </div>
        <svg
          aria-hidden="true"
          className="StyledIconBase-ea9ulj-0 lbJwfL sc-gKXOVf clsDmZ"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</section>
`;
